src/appleseedmaya/exporters/cameraexporter.cpp

```diff
@@ -161,11 +166,31 @@ void CameraExporter::createEntities(
         cameraParams.insert("shift_y", shift_y);
 
         // Maya's focal_length is given in mm so we convert it to meters.
-        cameraParams.insert("focal_length", cameraFn.focalLength() * 0.001);
+        cameraParams.insert("focal_length", cameraFn.focalLength() / cameraFn.cameraScale() * 0.001);
+        cameraParams.insert("horizontal_fov", asf::rad_to_deg(cameraFn.horizontalFieldOfView() * cameraFn.cameraScale()));
 
+        // Physical camera (thin-lens camera) parameter.
         if (dofEnabled)
         {
-            // TODO: extract dof params here...
+            bool autofocusEnabled = false;
+            if (AttributeUtils::get(node(), "asAutofocus", autofocusEnabled))
```

Comment:
There's a bug here: `AttributeUtils::get()` only return whether or not it has successfully retrieved the requested attribute, not the value of the attribute; here `autofocusEnabled` may be `false`.

Reply:
Ok, I changed it. Sorry that I have to ask again but it is not clear why it is really necessary. I followed the exact same construct as it is used for many attributes in renderglobalsnode.cpp, for instance
```cpp
bool prefilterSpikes;
    if (AttributeUtils::get(MPlug(globals, m_prefilterSpikes), prefilterSpikes))
    {
        frame->get_parameters().insert(
            "prefilter_spikes", prefilterSpikes);
    }
```
The same argument would apply here as well or not?

Reply:
Indeed, you're right.